test/new-e2e/tests/containers/k8s_test.go

```diff
@@ -797,28 +800,50 @@ func (suite *k8sSuite) TestSBOM() {
 			return
 		}
 
-		var sbomID string
-		for _, id := range sbomIDs {
-			if strings.HasPrefix(id, "ghcr.io/datadog/apps-nginx-server") {
-				sbomID = id
-				break
-			}
+		sbomIDs = lo.Filter(sbomIDs, func(id string, _ int) bool {
+			return strings.HasPrefix(id, "ghcr.io/datadog/apps-nginx-server")
+		})
+
+		// Can be replaced by require.NoEmptyf(…) once https://github.com/stretchr/testify/pull/1481 is merged
+		if !assert.NotEmptyf(c, sbomIDs, "No SBOM for ghcr.io/datadog/apps-nginx-server yet") {
+			return
 		}
+
+		images := lo.FlatMap(sbomIDs, func(id string, _ int) []*aggregator.SBOMPayload {
+			images, err := suite.Fakeintake.FilterSBOMs(id)
+			assert.NoErrorf(c, err, "Failed to query fake intake")
+			return images
+		})
+
 		// Can be replaced by require.NoEmptyf(…) once https://github.com/stretchr/testify/pull/1481 is merged
-		if !assert.NotEmptyf(c, sbomID, "No SBOM for ghcr.io/datadog/apps-nginx-server yet") {
+		if !assert.NotEmptyf(c, images, "No SBOM payload yet") {
 			return
 		}
 
-		images, err := suite.Fakeintake.FilterSBOMs(sbomID)
-		// Can be replaced by require.NoErrorf(…) once https://github.com/stretchr/testify/pull/1481 is merged
-		if !assert.NoErrorf(c, err, "Failed to query fake intake") {
+		images = lo.Filter(images, func(image *aggregator.SBOMPayload, _ int) bool {
+			return image.Status == sbom.SBOMStatus_SUCCESS
+		})
+
+		// Can be replaced by require.NoEmptyf(…) once https://github.com/stretchr/testify/pull/1481 is merged
+		if !assert.NotEmptyf(c, images, "No successful SBOM yet") {
 			return
 		}
+
+		images = lo.Filter(images, func(image *aggregator.SBOMPayload, _ int) bool {
+			cyclonedx := image.GetCyclonedx()
+			return cyclonedx != nil &&
+				cyclonedx.Metadata != nil &&
+				cyclonedx.Metadata.Component != nil &&
+				cyclonedx.Metadata.Component.Properties != nil
```

Comment:
I think a SBOM with a cyclone dx without properties should be an error case

Reply:
This is now asserted: https://github.com/DataDog/datadog-agent/pull/22118/files#diff-2bca1254c33d752a13e15a261a21b1282ec394f1e06fba2b8d850c4215cf5102R845